src/main/java/github/weichware10/util/db/Trials.java

```diff
@@ -0,0 +1,360 @@
+                dataBaseClient.schema,
+                new Timestamp(trialData.startTime.getMillis()),
+                trialData.getAnswer(),
+                trialData.trialId);
+
+        boolean success = false;
+
+        Connection conn = null;
+        Statement st = null;
+
+        try {
+            conn = DriverManager.getConnection(dataBaseClient.url, dataBaseClient.props);
+            st = conn.createStatement();
+            st.executeUpdate(query);
+            // DATAPOINTS setzen
+            dataBaseClient.datapoints.set(trialData.getData(), trialData.trialId);
+            success = true;
+        } catch (SQLException e) {
+            Logger.error("SQLException when executing setTrial", e);
+        } finally {
+            Util.closeQuietly(st);
+            Util.closeQuietly(conn);
+        }
+        return success;
+    }
+
+    /**
+     * Überprüft, ob trialID exisitiert und der Versuch dieser trialID bereits
+     * durchgeführt wurde.
+     *
+     * @param trialId - ID des Versuchs
+     * @return Verfügbarkeitsboolean
+     */
+    public boolean getAvailability(String trialId) {
+        // Benutzung von starttime um möglichst wenig Datenverbrauch zu erreichen
+        // timestamp hat meistens eine kleiner Größe als text
+        final String queryF = "SELECT starttime FROM %s.trials WHERE trialid LIKE '%s'";
+        final String query = String.format(queryF, dataBaseClient.schema, trialId);
+
+        boolean availabilty = false;
+
+        Connection conn = null;
+        Statement st = null;
+        ResultSet rs = null;
+
+        try {
+            conn = DriverManager.getConnection(dataBaseClient.url, dataBaseClient.props);
+            st = conn.createStatement();
+            rs = st.executeQuery(query);
+
+            // wenn es existiert, besteht Möglichkeit, dass availability true ist
+            if (rs.next()) {
+                // falls starttime == null ist -> true
+                availabilty = (rs.getString("starttime") == null) ? true : false;
+            }
+        } catch (SQLException e) {
+            Logger.error("SQLException when executing getTrialAvailability", e);
+        } finally {
+            Util.closeQuietly(rs);
+            Util.closeQuietly(st);
+            Util.closeQuietly(conn);
+        }
+
+        return availabilty;
+    }
+
+    /**
+     * Erstellt bestimmte Anzahl (amount) an Versuche (leer) mit der Konfiguration
+     * von configID.
+     *
+     * @param configId - ID der Konfiguration
+     * @param amount   - Anzahl der Versuche
+     * @return Liste mit vergebenen trialIDs
+     */
+    public List<String> add(String configId, int amount) {
+        // Überprüft, ob amount <= 0 ist
+        // oder Verfügbarkeit der Konfiguration mit der configId false ist
+        if (amount <= 0 || !dataBaseClient.configurations.getAvailability(configId)) {
+            return null;
+        }
+        final String query = """
```

Comment:
Fehlt hier nicht noch die final String queryF = """[...]""" in der der die SQL Anfrage steht?
Sodass die final String query = String.format(...); macht?

Reply:
das wird erst unten gemacht, da ja das ganze mit verschiedenen trialIds, mehrmals gemacht wird. Ich kann den Variablenname aber zu queryF ändern, damit das offensichtlicher ist